src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -235,11 +231,15 @@ class AuthScreens extends React.Component {
             return true;
         }
 
+        if (isTransitionOrNewWorkspacePage(this.props.currentURL) && !isTransitionOrNewWorkspacePage(nextProps.currentURL)) {
```

Comment:
Well, my reasoning here was since I added the `currentURL` Onyx key, that will change any time we navigate anywhere:

1. Opening settings modal
2. Switching reports
3. Etc...

And by default we probably don't want to re-render the whole component tree from `AuthScreens` down every time the route changes. The only time we _do_ want to force a re-render and trigger `componentDidUpdate` is when we are moving from `transition` or `workspace/new` to anywhere else, because that means it's now safe to load policies without risking overwriting a newly-created policy. Overall this design doesn't feel _great_, but I think it works.

If that all makes sense, I could add a comment like this:

```
/*
 * We don't want to re-render this component every time the `currentURL` prop changes,
 * but we do want to re-render when we're switching from `transition` or `workspace/new` to any other page.
 * Re-rendering in that case will re-trigger `componentDidUpdate` and `loadPoliciesBehindBeta`,
 * which we only want to do after we're done with the `transition` and `workspace/new` pages.
 * If we don't wait to load the policies until after we're done with those pages,
 * we may accidentally overwrite the newly-created policy and land on an infinite loading spinner. 
 */
```

Reply:
Another way we could maybe simplify this code without breaking the E/App philosophy would be to trigger `this.loadPoliciesBehindBeta` directly here in `shouldComponentUpdate`, but in general I'm hesitant to put side-effects in `shouldComponentUpdate`. Not sure I can explain why exactly, it just feels like a bad pattern. 🤷

Reply:
+1 to leaving the above comment for an explanation but I also think it could use a bit of re-working and additional lives to help with clarity. e.g.
> We don't want to re-render this component every time the `currentURL` prop changes **because xyz**